x-pack/plugin/wildcard/src/test/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldAggregationTests.java

```diff
@@ -0,0 +1,142 @@
+import org.elasticsearch.search.aggregations.support.AggregationInspectionHelper;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+public class WildcardFieldAggregationTests extends AggregatorTestCase {
+    private static final String WILDCARD_FIELD_NAME = "wildcard_field";
+    private static final int MAX_FIELD_LENGTH = 30;
+
+    private Optional<IndexWriterConfig> iwc;
+    private WildcardFieldMapper wildcardFieldMapper;
+    private WildcardFieldMapper.WildcardFieldType wildcardFieldType;
+
+
+    @Before
+    public void setup() {
+        // These settings are lifted out of WildcardFieldMapperTests
+        IndexWriterConfig iwc = newIndexWriterConfig(WildcardFieldMapper.WILDCARD_ANALYZER);
+        iwc.setMergePolicy(newTieredMergePolicy(random()));
+        this.iwc = Optional.of(iwc);
+
+        WildcardFieldMapper.Builder builder = new WildcardFieldMapper.Builder(WILDCARD_FIELD_NAME);
+        builder.ignoreAbove(MAX_FIELD_LENGTH);
+        wildcardFieldMapper = builder.build(new Mapper.BuilderContext(createIndexSettings().getSettings(), new ContentPath(0)));
+
+        wildcardFieldType = wildcardFieldMapper.fieldType();
+    }
+
+    private void addFields(ParseContext.Document parseDoc, Document doc, String docContent) throws IOException {
+        ArrayList<IndexableField> fields = new ArrayList<>();
+        wildcardFieldMapper.createFields(docContent, parseDoc, fields);
+
+        for (IndexableField indexableField : fields) {
+            doc.add(indexableField);
+        }
+    }
+
+    private void indexDoc(ParseContext.Document parseDoc, Document doc, RandomIndexWriter iw) throws IOException {
+        IndexableField field = parseDoc.getByKey(wildcardFieldMapper.name());
+        if (field != null) {
+            doc.add(field);
+        }
+        iw.addDocument(doc);
+    }
+
+    private void indexStrings(RandomIndexWriter iw, String... values) throws IOException {
+        Document doc = new Document();
+        ParseContext.Document parseDoc = new ParseContext.Document();
+        for (String value : values) {
+            addFields(parseDoc, doc, value);
+        }
+        indexDoc(parseDoc, doc, iw);
+    }
+
+    public void testTermsAggregation() throws IOException {
+        TermsAggregationBuilder aggregationBuilder = new TermsAggregationBuilder("_name")
+            .field(WILDCARD_FIELD_NAME)
+            .order(BucketOrder.key(true));
+
+        testCase(aggregationBuilder,
+            iwc, new MatchAllDocsQuery(),
+            iw -> {
+                indexStrings(iw, "a");
+                indexStrings(iw, "a");
+                indexStrings(iw, "b");
+                indexStrings(iw, "b");
+                indexStrings(iw, "b");
+                indexStrings(iw, "c");
+            },
+            (Consumer<StringTerms>) result -> {
```

Comment:
`(StringTerms results) ->` should do the same thing with a little less ceremony.